crates/fuel-core/src/database.rs

```diff
@@ -83,164 +82,89 @@ pub mod message;
 pub mod metadata;
 pub mod sealed_block;
 pub mod state;
-pub mod statistic;
 pub mod storage;
-pub mod transaction;
 pub mod transactions;
 
 #[derive(Clone, Debug)]
 pub struct Database<Description = OnChain>
 where
     Description: DatabaseDescription,
 {
-    data: StructuredStorage<DataSource<Description>>,
-    // used for RAII
-    _drop: Arc<DropResources>,
-}
-
-type DropFn = Box<dyn FnOnce() + Send + Sync>;
-#[derive(Default)]
-struct DropResources {
-    // move resources into this closure to have them dropped when db drops
-    drop: Option<DropFn>,
-}
-
-impl fmt::Debug for DropResources {
-    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        write!(f, "DropResources")
-    }
-}
-
-impl<F: 'static + FnOnce() + Send + Sync> From<F> for DropResources {
-    fn from(closure: F) -> Self {
-        Self {
-            drop: Option::Some(Box::new(closure)),
-        }
-    }
-}
-
-impl Drop for DropResources {
-    fn drop(&mut self) {
-        if let Some(drop) = self.drop.take() {
-            (drop)()
-        }
-    }
+    height: Arc<Mutex<Option<Description::Height>>>,
```

Comment:
I feel like since `height` is part of this, we could change the name of `Database` to something more specific and domain related. Will this always be the block height?

Reply:
The `Description` makes it kind of more domain-specific like `Database<OnChain>` or `Database<OffChain>`. I'm okay with improving naming if it is not good enough, do you have any suggestions?=)

Reply:
`Database<OnChain>` still doesn't help the reader understand what the database is used for, just that it for information that is on a (block)chain.

The fact that it is constrained to things that have a `height` implies that it isn't a general-purpose database.

Based on the traits implemented for the type, maybe `ChainStateDatabase`, `IterableChainStateDatabase`, or `AtomicIterableChainStateDatabase`? Sorry, I'm realizing I'm making a jab at inheritance design patterns :P
`ChainStateDatabase` probably is the most appropriate.

Reply:
I'm okay with renaming it to `ChainStateDatabase` or `HeightBasedDatabase` or `BlockchainDatabase`=)  But I will do that in a separate PR to not change a lot of files